saw-core-what4/src/Verifier/SAW/Simulator/What4/FirstOrder.hs

```diff
@@ -111,13 +113,30 @@ groundToFOV BaseRealRepr    _         = Left "Real is not FOV"
 groundToFOV BaseComplexRepr         _ = Left "Complex is not FOV"
 groundToFOV (BaseStringRepr _)      _ = Left "String is not FOV"
 groundToFOV (BaseFloatRepr _)       _ = Left "Floating point is not FOV"
-groundToFOV (BaseArrayRepr (Empty :> ty) b) _
-  | Right fot1 <- typeReprToFOT ty
-  , Right fot2 <- typeReprToFOT b
-  = pure $ FOVArray fot1 fot2
-groundToFOV (BaseArrayRepr _idx _b) _ = Left "Unsupported FOV Array"
+groundToFOV (BaseArrayRepr _ _) (ArrayMapping _) =
+    Left "Unsupported FOV Array (values only available by lookup)"
```

Comment:
As a general comment, we are assuming several things about the particular shapes of the arrays that we receive, but these assumptions are scattered throughout several places in the code. I think it would be helpful to have a central Note (see [here](https://github.com/GaloisInc/saw-script/blob/47a7632c973df3fdb2d32268edc92948fc150f2d/src/SAWScript/Crucible/Common/Override.hs#L234-L276) for prior art) that lists all of these assumptions, and then we can leave references to that Note in all the places that make use of these assumptions.

-----

Anyways, we are assuming here that the array will always be an `ArrayConcrete` and never an `ArrayMapping`. I think that's fine, since it's not obvious how to translate an `ArrayMapping` to a `FirstOrderValue`.

Reply:
I am adding a note in FiniteValue.hs (which seems like the best place for it)